src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -317,6 +318,50 @@ SyncCitusTableMetadata(Oid relationId)
 		ObjectAddressSet(relationAddress, RelationRelationId, relationId);
 		MarkObjectDistributed(&relationAddress);
 	}
+
+	CreateDependentViewsOnWorkers(relationId);
```

Comment:
We may want to move that call above the if check to have a better structure.

Reply:
the table needs to be marked as distributed, before we propagate the views. Because if the try to propagate the view before marking the table, at that time the table would be a regular(local) pg table, and citus would consider it as an unsupported dependency

Reply:
wouuld be nice to have a single line comment for this